plugins/playstation/src/playstation-header.vala

```diff
@@ -0,0 +1,87 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStationHeader : Object {
+  private File cue_file;
+  private File bin_file;
+	private string parent_directory;
+
+	private const size_t[] OFFSETS = { 0xD368, 0xD3A8 };
+	private const string[] IDS = { "SLUS", "SCUS", "SLES", "SCES", "SLPS", "SLPM", "SCPS" };
+
+	private size_t? header_offset;
+
+  public PlayStationHeader (File cue_file) {
+    this.cue_file = cue_file;
+  }
+
+  public void check_validity () throws PlayStationError {
+    check_cue_info ();
+
+		foreach (var offset in OFFSETS) {
+			var data = get_string_at_offset (offset, 50);
+			if (data.contains ("BOOT")) {
+				foreach (var id in IDS) {
+					if (data.contains (id)) {
+						string raw_id = data.split (id)[1].split (";")[0];
+						raw_id = raw_id.replace ("_", "-");
+						raw_id = raw_id.replace (".", "");
+						stdout.printf(id+raw_id + " - " + parent_directory + "\n");
+						return;
+					}
+				}
+			}
+		}
+		throw new PlayStationError.INVALID_SIZE (@"This is not a valid PlayStation file");
+  }
+
+	private void check_cue_info () {
+		try {
+			parent_directory = cue_file.get_parent ().get_path () + "/";
+      // Open file for reading and wrap returned FileInputStream into a
+      // DataInputStream, so we can read line by line
+      var dis = new DataInputStream (get_stream (cue_file));
+      string line;
+      if ((line = dis.read_line (null)) != null) {
+				var path = parent_directory + line.split("\"")[1];
+        bin_file = File.new_for_path (path);
+      }
+    } catch (Error e) {
+      error ("%s", e.message);
+    }
+	}
+
+  private FileInputStream get_stream (File file) throws PlayStationError {
+		try {
+			return file.read ();
+		}
+		catch (Error e) {
+			throw new PlayStationError.CANT_READ_FILE (@"Couldn't read file: $(e.message)");
+		}
+	}
+
+  private string get_string_at_offset (size_t offset, size_t size) throws PlayStationError {
```

Comment:
What about renaming this method `get_string_at_offset_for_length()` (renaming `size` to `length`) and having another one named `get_string_at_offset()` which doesn't take a `size` param but instead read char after char until it reaches the null char... or any other way to get a variable length string. It would allow to get the full header string.

Or instead don't touch this one and add a `get_line_at_offset` which reads until reaching the \0 or the \n char.